lib/kodama/client.rb

```diff
@@ -5,6 +5,17 @@
 require 'uri'
 
 module Kodama
+  # A wrapper exception raised when an exception is raised from within the
+  # transaction block.  When this exception is raised, the next execution of the
+  # client may not start from the correct binlog position.
+  class TransactionError < RuntimeError
```

Comment:
Not particular reason other than a transaction error is a kind of runtime error.

TransactionError is meant to wrap any Exception.  In the sense, it may make the most sense for the class to inherit Exception.  But regardless of the parent class, the wrapper exception approach is not backward compatible since it changes the type of exception to be raised.

To avoid this, adding an attribute to indicate if it's a transaction error or not to the original exception may be a better approach.